subgraphs/playback/src/utils/mocks.ts

```diff
@@ -1,5 +1,190 @@
-export const mocks = {
-  User: () => ({
-    displayName: 'Mocked Watson'
-  })
-}
\ No newline at end of file
+import { Spotify } from '../dataSources/spotify.types';
+import { SpotifyDataSource } from '../dataSources/spotify';
+
+const userState: {
+  [userId: string]: {
+    actions: { disallows: [string] };
+    context: {
+      external_urls: { spotify: string };
+      href: string;
+      type: 'artist' | 'playlist' | 'album' | 'show';
+      uri: string;
+    };
+    device: {
+      id: string;
+      is_active: boolean;
+      is_private_session: boolean;
+      is_restricted: boolean;
+      name: string;
+      type: string;
+      volume_percent: number;
+    };
+    is_playing: boolean;
+    item: { id: string };
+    progress_ms: number;
+    repeat_state: 'context' | 'off' | 'track';
+    shuffle_state: boolean;
+    timestamp: number;
+    currently_playing_type: string;
+  };
+} = {
+  default: {
```

Comment:
Since JS objects are pass-by-reference, you may find that changing state for one user modifies that state for everyone since its just assigning `userState[userId]` from this object. To see what I mean, put the following in the browser JS console:

```ts
const foo = {
  a: 1
}
const bar = foo;

bar.a = 2;

console.log(foo.a) // => 2
```

Instead, it might be better to create a function that returns a _new_ object each time you call it. Try something like this:

```ts
const userState: {
  [userId: string]: {...}
} = {}

function createUserState() {
  return {
    // all the contents of `default` now
  }
}

function addUser(userId: string) {
  // ...
  userState[userId] = createUserState();
}
```

This will make sure that each user state slice gets its own object reference, rather than sharing all of them from the `default` object.

I'd also just recommend calling this `findOrCreateUserState` and returning the object here:

```ts
function findOrCreateUserState(userId: string) {
  // I _think_ you can use the logical OR assignment here to shorten to a single line:
  // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_OR_assignment
  return userState[userId] ||= createUserState()
}
```

This function also guarantees you have a `userState` object, so this should eliminate the need for you to do `if (userState[userId])` checks everywhere.

Reply:
This has now been implemented.